src/sentry/middleware/health.py

```diff
@@ -19,9 +21,16 @@ def process_request(self, request):
 
         from sentry.status_checks import check_all
         from sentry.utils import json
-        problems, checks = check_all()
 
+        results = check_all()
+        problems = list(itertools.chain.from_iterable(results.values()))
         return HttpResponse(json.dumps({
```

Comment:
Not overly concerned about compat here. Just that this endpoint has a very specific purpose and may not need to change.